### ISSUE TITLE
`multiple_inherent_impl` shouldn't trigger for `#[cfg(test)] impl`

### ISSUE THREAD

**Issue by nickdrozd** [rust-lang/rust-clippy#13040]
### Summary

I have a struct with `impl` block, plus an additional `impl` block for testing:

```rust
struct X {}

impl X {}

#[cfg(test)]
impl X {}
```

The `multiple_inherent_impl` lint flags this, but it shouldn't. Combining the two `impl` blocks would cause a change in behavior, namely enabling the test methods outside of the test environment.

### Lint Name

multiple_inherent_impl

### Version

```text
rustc 1.81.0-nightly (6292b2af6 2024-07-02)
binary: rustc
commit-hash: 6292b2af620dbd771ebb687c3a93c69ba8f97268
commit-date: 2024-07-02
host: x86_64-unknown-linux-gnu
release: 1.81.0-nightly
LLVM version: 18.1.7
```

**Comment by Jarcho** [comment]
It shouldn't trigger if there are any attributes, not just `#[cfg(test)]`.

**Comment by nickdrozd** [comment]
I guess in general it should trigger only for two `impl` blocks with identical attributes. So for example, for two `#[cfg(test)] impl` blocks it should trigger.

**Comment by profetia** [comment]
Looks like this already fixed in the latest nightly.

**Comment by vogelbirb** [comment]
Looks like this doesn't trigger with `#[cfg(test)] impl` on Clippy 0.1.89, but it seems like it still does for other attributes, unlike what @Jarcho suggested. 
https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=da8219de8517cd7cebf48f6ee7c5f413

**Comment by Jarcho** [comment]
My comment was that this attribute shouldn't be singled out, not about what the lint currently does.

**Comment by MikkelPaulson** [comment]
@mrchilliballs:
> Looks like this doesn't trigger with `#[cfg(test)] impl` on Clippy 0.1.89, but it seems like it still does for other attributes, unlike what **Jarcho** suggested. https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=da8219de8517cd7cebf48f6ee7c5f413

Clippy doesn't evaluate `#[cfg(test)]` code at all unless run with `--tests`, which is not the case in the playground. Evaluating the same code using `cargo clippy --tests` in Rust 1.90.0 does indeed produce a warning.
